resources/ext/cli/config-migration

```diff
@@ -0,0 +1,244 @@
+do
+  case $opt in
+    f)
+      force=true ;;
+    h)
+      usage ;;
+    *)
+      usage ;;
+  esac
+done
+
+${force:=false}
+
+orig_config_dir=/tmp/puppetdb-orig-configs
+orig_jetty_file=${orig_config_dir}/jetty.ini
+orig_database_file=${orig_config_dir}/database.ini
+orig_config_file=${orig_config_dir}/config.ini
+orig_ini_files=($orig_jetty_file $orig_database_file $orig_config_file)
+
+config_dir=/etc/puppetlabs/puppetdb/conf.d
+database_file=${config_dir}/database.ini
+jetty_file=${config_dir}/jetty.ini
+config_file=${config_dir}/config.ini
+ini_files=($config_file $jetty_file $database_file)
+
+# Deal with config files
+echo "Migrating configuration files ${orig_ini_files[*]} to ${config_dir}"
+if [ -e '/tmp/puppetdb-upgrade-config-files.tgz' ] ; then
+  mkdir -p $orig_config_dir
+  tar -xzf '/tmp/puppetdb-upgrade-config-files.tgz' -C $orig_config_dir
+else
+  echo "Config archive not found. Not proceding with migration"
+  exit 0
+fi
+for orig_ini_file in "${orig_ini_files[@]}" ; do
+  migrate "$orig_ini_file" "$config_dir"
+done
+
+echo "Checking to see if we need to update default settings"
+for ini_file in "${ini_files[@]}"; do
+  if [ -f "${ini_file}" ] ; then
+    # Check settings are correct and fix or warn
+    orig_settings=(
+      "vardir:/var/lib/puppetdb"
+      "logging-config:/etc/puppetdb/logback.xml"
+      "subname:file:/var/lib/puppetdb/db/db;hsqldb.tx=mvcc;sql.syntax_pgs=true"
+      "ssl-key:/etc/puppetdb/ssl/private.pem"
+      "ssl-cert:/etc/puppetdb/ssl/public.pem"
+      "ssl-ca-cert:/etc/puppetdb/ssl/ca.pem"
+    )
+    new_settings=(
+      "vardir:/opt/puppetlabs/server/data/puppetdb"
+      "logging-config:/etc/puppetlabs/puppetdb/logback.xml"
+      "subname:file:/opt/puppetlabs/server/data/puppetdb/db/db;hsqldb.tx=mvcc;sql.syntax_pgs=true"
+      "ssl-key:/etc/puppetlabs/puppetdb/ssl/private.pem"
+      "ssl-cert:/etc/puppetlabs/puppetdb/ssl/public.pem"
+      "ssl-ca-cert:/etc/puppetlabs/puppetdb/ssl/ca.pem"
+    )
+
+    for i in "${orig_settings[@]}"; do
+      setting="${i%%:*}"
+      orig_value="${i#*:}"
+      for j in "${new_settings[@]}"; do
+        if [ "${j%%:*}" == "${setting}" ] ; then
+          new_value="${j#*:}"
+          if grep -qe "^${setting}" "$ini_file"; then
+            if grep -qe "^${setting}[[:space:]]*=[[:space:]]*${orig_value}$" "$ini_file"; then
+              if $force; then
+                replaceline "^${setting}.*" "${setting} = ${new_value}" "$ini_file"
+                echo "Updated setting ${setting} in ${ini_file}."
+                case "$setting" in
+                  vardir|ssl-*)
+                    migrate "$orig_value" "$new_value"
+                    ;;
+                  subname)
+                    orig_value="${orig_value#file:}"
+                    orig_value="${orig_value%/db;hsqldb.tx=mvcc;sql.syntax_pgs=true}"
+                    new_value="${new_value#file:}"
+                    new_value="${new_value%/db;hsqldb.tx=mvcc;sql.syntax_pgs=true}"
```

Comment:
@kbarber I've updated this to pull out the last db from the path. It looks like it's working now. I also rebased against stable